causal_testing/testing/causal_test_engine.py

```diff
@@ -148,63 +146,17 @@ def _return_causal_test_results(self, estimator, causal_test_case):
-                confidence_intervals=confidence_intervals,
-            )
-        elif causal_test_case.estimate_type == "risk_ratio":
-            logger.debug("calculating risk_ratio")
-            risk_ratio, confidence_intervals = estimator.estimate_risk_ratio(**causal_test_case.estimate_params)
-
-            causal_test_result = CausalTestResult(
-                estimator=estimator,
-                test_value=TestValue("risk_ratio", risk_ratio),
-                effect_modifier_configuration=causal_test_case.effect_modifier_configuration,
-                confidence_intervals=confidence_intervals,
-            )
-        elif causal_test_case.estimate_type == "coefficient":
-            logger.debug("calculating coefficient")
-            coefficient, confidence_intervals = estimator.estimate_unit_ate()
-            causal_test_result = CausalTestResult(
-                estimator=estimator,
-                test_value=TestValue("coefficient", coefficient),
-                effect_modifier_configuration=causal_test_case.effect_modifier_configuration,
-                confidence_intervals=confidence_intervals,
-            )
-        elif causal_test_case.estimate_type == "ate":
-            logger.debug("calculating ate")
-            ate, confidence_intervals = estimator.estimate_ate()
-            causal_test_result = CausalTestResult(
-                estimator=estimator,
-                test_value=TestValue("ate", ate),
-                effect_modifier_configuration=causal_test_case.effect_modifier_configuration,
-                confidence_intervals=confidence_intervals,
-            )
-            # causal_test_result = CausalTestResult(minimal_adjustment_set, ate, confidence_intervals)
-            # causal_test_result.apply_test_oracle_procedure(self.causal_test_case.expected_causal_effect)
-        elif causal_test_case.estimate_type == "ate_calculated":
-            logger.debug("calculating ate")
-            ate, confidence_intervals = estimator.estimate_ate_calculated()
-            causal_test_result = CausalTestResult(
-                estimator=estimator,
-                test_value=TestValue("ate", ate),
-                effect_modifier_configuration=causal_test_case.effect_modifier_configuration,
-                confidence_intervals=confidence_intervals,
-            )
-            # causal_test_result = CausalTestResult(minimal_adjustment_set, ate, confidence_intervals)
-            # causal_test_result.apply_test_oracle_procedure(self.causal_test_case.expected_causal_effect)
-        else:
-            raise ValueError(
-                f"Invalid estimate type {causal_test_case.estimate_type}, expected 'ate', 'cate', or 'risk_ratio'"
-            )
+        if not hasattr(estimator, f"estimate_{causal_test_case.estimate_type}"):
+            raise NotImplementedError(f"{estimator.__class__} has no {causal_test_case.estimate_type} method.")
```

Comment:
`NotImplementedError` I think is reserved for abstract methods. Here an `AttributeError` would be more appriopriate

Reply:
Fixed